src/java.base/share/classes/sun/security/rsa/RSACore.java

```diff
@@ -117,8 +117,8 @@ public static byte[] convert(byte[] b, int ofs, int len) {
      */
     public static byte[] rsa(byte[] msg, RSAPublicKey key)
             throws BadPaddingException {
-        if (useNativeRsa && key instanceof sun.security.rsa.RSAPublicKeyImpl) {
-            byte[] ret = NativeRSACore.rsa(msg, (sun.security.rsa.RSAPublicKeyImpl) key);
+        if (useNativeRsa && key instanceof sun.security.rsa.RSAPublicKeyImpl rsaKey && NativeCrypto.isLoaded()) {
```

Comment:
Why doesn't this test (and the similar test on line 149) not call `NativeCrypto.isNativeCryptoSafe()`?

Reply:
Moved `InternalCRIUSupport.isCheckpointAllowed()` into `getVersion()`, and removed `isNativeCryptoSafe()`.